hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/HTSPrecompiledContract.java

```diff
@@ -740,6 +740,38 @@ yield switch (nestedFunctionSelector) {
                             syntheticTxnFactory,
                             infrastructureFactory,
                             precompilePricingUtils);
+                    case AbiConstants.ABI_ID_TRANSFER_FROM -> checkFeatureFlag(
+                            dynamicProperties.areAllowancesEnabled(),
+                            () ->
+                                    new ERCTransferPrecompile(
```

Comment:
I wonder if we should look into a HTSPrecompiledContract local Dagger Module that can produce these objects.  There are a lot of constructor params that are just passed directly from the constructor to the instance.

possibly https://dagger.dev/dev-guide/assisted-injection

Not for this PR.